pkg/database/connection.go

```diff
@@ -18,8 +18,9 @@ import (
 
 // Database Access Object. Use a DAO instance so we can replace the pool object in the unit tests.
 type DAO struct {
-	pool      pgxpoolmock.PgxPool
-	batchSize int
+	DBInitialized bool
```

Comment:
1. Change to `dbConnInitialized` to avoid confusion with the function where we create the tables.
2. Change to private and add a function to access it. This will prevent us from changing the value elsewhere.